src/transformers/testing_utils.py

```diff
@@ -1570,6 +1572,35 @@ def LoggingLevel(level):
         transformers_logging.set_verbosity(orig_level)
 
 
+class TemporaryHubRepo:
+    """Create a temporary Hub repository and return its `RepoUrl` object. This is similar to
+    `tempfile.TemporaryDirectory` and can be used as a context manager.  For example:
+
+        with TemporaryHubRepo() as temp_repo:
+            ...
+
+    Upon exiting the context, the repository and everything contained in it are removed.
+
+    Example:
+
+    ```python
+    with TemporaryHubRepo() as temp_repo:
+        model.push_to_hub(tmp_repo.repo_id, token=self._token)
+    ```
+    """
+
+    def __init__(self, token=None):
+        self.token = token
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            self.repo_url = huggingface_hub.create_repo(Path(tmp_dir).name, token=self.token)
+
+    def __enter__(self):
+        return self.repo_url
+
+    def __exit__(self, exc, value, tb):
+        delete_repo(repo_id=self.repo_url.repo_id, token=self.token, missing_ok=True)
```

Comment:
I just realized that I'm not sure anymore if `__exit__` is called when a test fails :confused: Might be the case but not 100% sure. If not called, it means tmp repos won't be garbage collected as previously with the "try/finally".

Reply:
It will!  Thank god the python core members implement this !